src/analysis/calculus/uniform_limits_deriv.lean

```diff
@@ -0,0 +1,394 @@
+/-
+Copyright (c) 2022 Kevin H. Wilson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin H. Wilson
+-/
+import analysis.calculus.fderiv
+import analysis.calculus.mean_value
+import analysis.convex.topology
+import analysis.normed_space.is_R_or_C
+import data.complex.is_R_or_C
+import order.filter.basic
+
+/-!
+# Swapping limits and derivatives via uniform convergence
+
+The purpose of this file is to prove that the derivative of the pointwise limit of a sequence of
+functions is the pointwise limit of the functions' derivatives when the derivatives converge
+_uniformly_. The formal statement appears as `has_fderiv_at_of_tendsto_uniformly_on`.
+
+## Main statements
+
+* `has_fderiv_at_of_tendsto_uniformly_on` : If `f : ℕ → E → G` is a sequence of functions with
+  derivatives `f' : ℕ → (E → (E →L[𝕜] G))` and the `f` converge pointwise to `g` and the `f'`
+  converge _uniformly_ on some closed ball, then the derivative of `g'` is the pointwise limit
+  of the `f'` on the closed ball
+
+## Implementation notes
+
+Our proof utilizes three major components:
+  * `convex.norm_image_sub_le_of_norm_has_fderiv_within_le`: The mean value inequality for
+    vector-valued functions over `ℝ` and `ℂ`
+  * `norm_add_le`: The triangle inequality
+  * `uniform_cauchy_seq_on.tendsto_uniformly_on_of_tendsto` which allows us to upgrade pointwise
+    convergence to uniform convergence by showing that the Cauchy sequences converge uniformly to 0
+
+## Tags
+
+uniform convergence, limits of derivatives
+-/
+
+open finset filter metric
+open_locale uniformity filter
+
+section convenience_lemmas
+
+/-- Convenience lemma -/
+private lemma mul_cancel_inv_left₀ {a b : ℝ} (ha : a ≠ 0) : a⁻¹ * (a * b) = b :=
```

Comment:
[`inv_mul_cancel_left₀`](https://leanprover-community.github.io/mathlib_docs/algebra/group_with_zero/basic.html#inv_mul_cancel_left%E2%82%80)

We frown upon private lemmas like this. They either should be in the appropriate place in the library, or not exist at all. In this case it already exists.

Reply:
Ah! Messed up the convention and couldn't find it. Interestingly, `simp` didn't work for me when I was doing this even though it's a `simp` lemma. Perhaps I mistyped :-/

Reply:
You need `simp*` or `simp [ha]` for `simp` to do anything: it needs to use `ha`.